centreon/src/Core/Service/Infrastructure/Repository/DbWriteServiceActionLogRepository.php

```diff
@@ -0,0 +1,266 @@
+                $serviceId,
+                $newService->getName(),
+                ActionLog::ACTION_TYPE_ADD,
+                $this->contact->getId()
+            );
+            $actionLogId = $this->writeActionLogRepository->addAction($actionLog);
+            $actionLog->setId($actionLogId);
+
+            $details = $this->getServicePropertiesAsArray($newService);
+            $this->writeActionLogRepository->addActionDetails($actionLog, $details);
+
+            return $serviceId;
+        } catch (\Throwable $ex) {
+            $this->error($ex->getMessage(), ['trace' => $ex->getTraceAsString()]);
+
+            throw $ex;
+        }
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function update(Service $service): void
+    {
+        try {
+            $currentService = $this->readServiceRepository->findById($service->getId());
+            if ($currentService === null) {
+                throw new RepositoryException(sprintf('Cannot find service to update (ID: %d).', $service->getId()));
+            }
+
+            $currentServiceDetails = $this->getServicePropertiesAsArray($currentService);
+            $updatedServiceDetails = $this->getServicePropertiesAsArray($service);
+            $diff = array_diff_assoc($updatedServiceDetails, $currentServiceDetails);
+
+            $this->writeServiceRepository->update($service);
+
+            $actionsToLog = [];
+
+            if (array_key_exists('isActivated', $diff)) {
+                $actionType = (bool) $diff['isActivated']
+                    ? ActionLog::ACTION_TYPE_ENABLE
+                    : ActionLog::ACTION_TYPE_DISABLE;
+
+                $actionsToLog[] = new ActionLog(
+                    self::SERVICE_OBJECT_TYPE,
+                    $service->getId(),
+                    $service->getName(),
+                    $actionType,
+                    $this->contact->getId()
+                );
+
+                unset($diff['isActivated']);
+            }
+
+            if (count($diff) > 0) {
+                $actionsToLog[] = new ActionLog(
+                    self::SERVICE_OBJECT_TYPE,
+                    $service->getId(),
+                    $service->getName(),
+                    ActionLog::ACTION_TYPE_CHANGE,
+                    $this->contact->getId()
+                );
+            }
+
+            foreach ($actionsToLog as $actionLog) {
+                $actionLogId = $this->writeActionLogRepository->addAction($actionLog);
+                if ($actionLogId === 0) {
+                    throw new RepositoryException('Action log ID cannot be 0');
+                }
+                $actionLog->setId($actionLogId);
+
+                if ($actionLog->getActionType() === ActionLog::ACTION_TYPE_CHANGE) {
+                    $this->writeActionLogRepository->addActionDetails($actionLog, $diff);
+                }
+            }
+        } catch (\Throwable $ex) {
+            $this->error($ex->getMessage(), ['trace' => $ex->getTraceAsString()]);
```

Comment:
you can specify business information with $service in the context of log to have more informations from the log

Reply:
Done

Reply:
Yes it's good, thanks for your change ;)